examples/keras_recipes/near_dup_parser.py

```diff
@@ -0,0 +1,538 @@
+that are worth checking out in this regard:
+
+* [Metric learning for image similarity search](https://keras.io/examples/vision/metric_learning)
+* [Image similarity estimation using a Siamese Network with a triplet loss](https://keras.io/examples/vision/siamese_network)
+
+Finally, this example uses the following resource as a reference and as such reuses code
+from there:
+[Locality Sensitive Hashing for Similar Item Search](https://towardsdatascience.com/locality-sensitive-hashing-for-music-search-f2f1940ace23).
+
+_Note that in order to optimize the performance of our parser, you should be connected to
+a GPU._
+"""
+
+"""
+## Imports
+"""
+
+import matplotlib.pyplot as plt
+import tensorflow as tf
+import numpy as np
+import time
+
+import tensorflow_datasets as tfds
+
+tfds.disable_progress_bar()
+
+"""
+## Load dataset and create a training set of 1000 images
+
+To keep the runtime of the example short, we will be using a subset of 1000 images from
+the `tf_flowers` dataset (available through
+[TensorFlow Datasets](https://www.tensorflow.org/datasets/catalog/tf_flowers))
+to build our vocabulary.
+"""
+
+train_ds, validation_ds = tfds.load(
+    "tf_flowers", split=["train[:85%]", "train[85%:]"], as_supervised=True
+)
+
+IMAGE_SIZE = 224
+NUM_IMAGES = 1000
+
+images = []
+labels = []
+
+for (image, label) in train_ds.take(NUM_IMAGES):
+    image = tf.image.resize(image, (IMAGE_SIZE, IMAGE_SIZE))
+    images.append(image.numpy())
+    labels.append(label.numpy())
+
+images = np.array(images)
+labels = np.array(labels)
+
+"""
+## Pre-trained model
+"""
+
+"""
+In this section, we load our image classification model that was trained on the
+`tf_flowers` dataset. 85% of the total images were used to build the training set. For
+more details on the training, you check out
+[this notebook](https://github.com/sayakpaul/near-dup-parser/blob/main/bit-supervised-training.ipynb).
+
+The underlying model is a BiT-ResNet (proposed in
+[Big Transfer (BiT): General Visual Representation Learning](https://arxiv.org/abs/1912.11370)).
+BiT-ResNet family of models are known to provide excellent transfer performance across
+a wide variety of different downstream tasks. 
+"""
+
+"""shell
+!wget -q https://git.io/JuV8G
+!tar xf flower_model_bit_0.96875.gz
+"""
+
+bit_model = tf.keras.models.load_model("flower_model_bit_0.96875")
+bit_model.count_params()
+
+"""
+## Create an embedding model
+
+To parse the similar images given a query image, we need to first generate vector
+representations of the images. For this purpose, we will be branching the above image
+classifier such that it becomes apt for our purpose.
+"""
+
+embedding_model = tf.keras.Sequential(
+    [
+        tf.keras.layers.InputLayer((IMAGE_SIZE, IMAGE_SIZE, 3)),
+        tf.keras.layers.experimental.preprocessing.Rescaling(scale=1.0 / 255),
+        bit_model.layers[1],
+        tf.keras.layers.Lambda(tf.math.l2_normalize),
```

Comment:
Here, you can use a `Normalization(mean=0, variance=1)` layer.  Using `l2_normalize` without specifying an axis will normalize the entire batch, which is not what you intend.

Reply:
Thanks for spotting this!